lib/pure/hashes.nim

```diff
@@ -123,6 +123,33 @@ proc hiXorLo(a, b: uint64): uint64 {.inline.} =
     else:
       result = hiXorLoFallback64(a, b)
 
+when defined(js):
+  import std/jsbigints
+  import std/private/jsutils
+
+  proc noBigInt(): bool =
+    asm """`result` = typeof BigInt == 'undefined'"""
+
+  proc hiXorLoJs(a, b: JsBigInt): JsBigInt =
+    let prod = a * b
+    let mask = (big"1" shl big"64") - big"1"
+    result = (prod shr big"64") xor (prod and mask)
```

Comment:
i suspect this would be more efficient (but runnableExamples suggestion from https://github.com/nim-lang/Nim/pull/16863/files#r567099096 still holds):

```suggestion
    result = (prod shr big"64") xor prod.wrapToUint(64)
```

but we need to ensure this is valid even for signed `prod` (can be checked in tjsbigints)

Reply:
could be tested in the future.